helm/cluster/values.schema.json

```diff
@@ -744,6 +744,12 @@
                         "replicas"
                     ],
                     "properties": {
+                        "apiServerPort": {
+                            "type": "integer",
+                            "title": "API server port",
+                            "description": "The public facing API server Load Balancer port.",
```

Comment:
Maybe we should be more concrete on the load balancer thing. One could confuse this with the API server port.

Reply:
I guess we can call it `apiServerLoadBalancerPort` then? The whole thing that's being set (i.e. `spec.ClusterNetwork.APIServerPort` in the Cluster CR) is very confusing to begin with and is used inconsistently, although I've only tested it on CAPA and just viewed the code for the other providers. Let me explain:
### CAPI
CAPI specifies this field as ["the port the API Server should bind to"](https://github.com/kubernetes-sigs/cluster-api/blob/1b662d33f8fe55511b47e0e8f9044c77acb848e4/api/v1beta1/cluster_types.go#L351-L354), however this isn't really used anywhere in the code as far as I can tell. The actual port that's being used comes from the KubeadmConfig. 

### CAPA
In CAPA (where I want to use this) the value is used to [configure the load balancer port **only**](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f8b9cfd43ff0dd3e2887c3acad3d1ac53284cf8c/pkg/cloud/services/elb/loadbalancer.go#L962-L969) (set [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f8b9cfd43ff0dd3e2887c3acad3d1ac53284cf8c/pkg/cloud/scope/cluster.go#L290-L295)). The destination port (i.e. the bindPort of the apiserver) is hardcoded to 6443. (Side note this is only for "classic" LBs, all others are hardcoded, but it's the default and is what we use anyway)

### CAPZ
CAPZ on the other hand sets [both the LB's port and the destination port ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/eadf744eb9ebbb7efa91bfed915104a23ed0abc8/azure/services/loadbalancers/spec.go#L229-L230C5). So I guess this means that you also need to set the `bindPort` when configuring the KubeadmControlPlane, but like I said I haven't tested this on anything other than CAPA.

### CAPV and CAPVCD

The value for `Spec.ClusterNetwork.APIServerPort` appears unused in those.

Reply:
Yeah, sounds confusing. I'd stick to how it is right now then.